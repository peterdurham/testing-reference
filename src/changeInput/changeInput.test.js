import React from "react";
import ChangeInput from "./changeInput";
import { render, fireEvent, cleanup } from "@testing-library/react";

afterEach(cleanup);

test("displays the correct greeting", () => {
  const { getByLabelText, getByTestId } = render(<ChangeInput />);
  const input = getByLabelText("user-name");
  const greeting = getByTestId("change-input-greeting");
  expect(input.value).toBe("");
  expect(greeting.textContent).toBe("Welcome, Anonymous User!");
  fireEvent.change(input, { target: { value: "Rafael" } });
  expect(input.value).toBe("Rafael");
  expect(greeting.textContent).toBe("Welcome, Rafael!");
});

test("FocusInput matches snapshot", () => {
  const { container } = render(<ChangeInput />);
  expect(container.firstChild).toMatchSnapshot();
});
