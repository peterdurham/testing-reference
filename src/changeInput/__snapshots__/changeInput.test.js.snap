// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FocusInput matches snapshot 1`] = `
<div>
  <span
    data-testid="change-input-greeting"
  >
    Welcome, 
    Anonymous User
    !
  </span>
  <input
    aria-label="user-name"
    placeholder="Your name"
    type="text"
    value=""
  />
</div>
`;
